src/ex.c

```diff
@@ -966,8 +967,38 @@ static VbCmdResult ex_handlers(Client *c, const ExArg *arg)
 
 static VbCmdResult ex_shortcut(Client *c, const ExArg *arg)
 {
-    /* TODO no implemented yet */
-    return CMD_SUCCESS;
+    char *key, *uri;
+    gboolean res = CMD_ERROR;
+
+    if (!c) {
+        return CMD_ERROR;
+    }
+
+    if (!arg || !arg->name) {
+        return CMD_ERROR;
+    }
+
+    if (!arg->rhs || !arg->rhs->str || !*arg->rhs->str) {
+        return CMD_ERROR;
+    }
+
+    if (strstr(arg->name, "shortcut-add")) {
```

Comment:
Thanks a lot for the hints, I learned a lot. I will push another update to align the implementation to the master branch's version.

Though, I would like to keep the stripping where it is, here is why:

1. We support ex commands such as ```set a=1``` and ```set     a = 1```, and they do exaclty the same (https://github.com/fanglingsu/vimb/blob/webkit2/src/ex.c#L910). I wanted to have the same functionality for ```shortcut-add```.
2. Allowing spaces after shortcut keys does not work as ```shortcut_lookup()``` splits key and value on the first space (https://github.com/fanglingsu/vimb/blob/master/src/shortcut.c#L197). Also tailing spaces on keys don't make any sense to me.
3. Moving ```g_strstrip()``` to ```shortcut.c``` would require changing the shortcut API as key and value paramters are ```const char*``` over there.